### PR TITLE
Help  PHPStorm to resolve types from the container

### PR DESCRIPTION
**Pull request by mitogh**
Allow PHPStorm Editor to resolve the different types when using the
container to get an instance to an object, this works only when the
container is called with a string that is either the full quailified
class of the object or when the name is used on the container such as:

```php
tribe( Object_Type::class )->resolves_names_of_method_or_properties();
tribe( 'Object_Type' )->resolves_names_of_method_or_properties();
```

### CODE REVIEW

**Comment by sc0ttkclark** [comment]
Wow, wicked!

**Review by sc0ttkclark — APPROVED** [review]

File: .phpstorm.meta.php [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace PHPSTORM_META {
+	// Allow PHP Storm Editor to resolve return types when calling tribe( Object_Type::class ) or tribe( `Object_type` )
```
Comment (sc0ttkclark) [inline .phpstorm.meta.php:4]:
```suggestion
	// Allow PhpStorm IDE to resolve return types when calling tribe( Object_Type::class ) or tribe( `Object_type` )
```

**Comment by sc0ttkclark** [comment]
This needs to ultimately be added to all plugin repos, correct?

**Comment by mitogh** [comment]
> This needs to ultimately be added to all plugin repos, correct?

Not really as long as the plugin uses common the `.phpstorm.meta.php` file is recursively added  so if you have a specif configuration on other plugins those are merged into a single one.

**Comment by sc0ttkclark** [comment]
Oh wow, interesting! Much wow, much autocomplete!

**Comment by mitogh** [comment]
Oh for sure it was a pain to describe or create a variable just to get the autocomplete. 

Reference: https://www.jetbrains.com/help/phpstorm/ide-advanced-metadata.html

**Review by bordoni — DISMISSED** [review]

**Pull request merged**